lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -468,7 +468,7 @@ def store_at_0
           end
 
           def update_positions
-            old_position = send("#{position_column}_was").to_i
+            old_position = send("#{position_column}_was") || bottom_position_in_list + 1
```

Comment:
I would say it's definitely needed. Without `to_i` old_position will be set as nil which will cause error in `shuffle_positions_on_intermediate_items` method which does comparison `old_position < new_position`. 
I guess that `to_i` might have been originaly used to patch this error since it converts nil to 0. But in that case we have this reordering issue. When nil is converted to 0, in the `shuffle_positions_on_intermediate_items` old_position is considered to be lower than new_position, so it assumes that element that was nil was closer to the top of the list and as a result moves old element closer to the top (so we would have [0, 1] instead of [1, 2]). 
After this tweak problem is solved, since if position is nil we simulate a situation when it is at the bottom of the list, so `shuffle_positions_on_intermediate_items` is feeded with proper data that allows all items **below** specific position be properly reordered.
I hope I clarified it enough, I did my best at least. Let me know if you need anything else.